python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -738,6 +742,26 @@ def inner(fn: Callable[..., Any]) -> AssetsDefinition:
             **check_outs_by_output_name,
         }
 
+        if specs:
```

Comment:
I had to rearrange this function a bit to make sure that the values were computed in the right order (i.e. we need to have internal_deps before we can generate the "corrected" set of asset ins, and we need to have the set of asset ins before we create the op)